### ISSUE TITLE
[A11Y] Registration form fields are not associated with the respective labels 

### ISSUE THREAD

**Issue by foxbunny** [indico/indico#6059]
**Describe the bug**

The form fields on the registration page are not associated with the respective labels. This includes (but not limited to):

**To Reproduce**
1. Go to a registration form using a screen reader
2. Focus any control
3. Observe that the field name is not announced

**Expected behavior**

The field name is announced.

**Additional context**

https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html
